internal/plugintest/working_dir.go

```diff
@@ -169,15 +171,64 @@ func (wd *WorkingDir) planFilename() string {
 	return filepath.Join(wd.baseDir, PlanFileName)
 }
 
-// CreatePlan runs "terraform plan" to create a saved plan file, which if successful
-// will then be used for the next call to Apply.
-func (wd *WorkingDir) CreatePlan(ctx context.Context) error {
+type CreatePlanResponse struct {
+	Diagnostics []tfjson.Diagnostic
+	Stdout      string
+}
+
+// CreatePlan runs CreatePlanJSON first and will fall back to running terraform plan if an error
+// is returned from CreatePlanJSON indicating that the version of Terraform that is being used
+// does not support the `-json` flag with "terraform plan". CreatePlan will create a saved plan
+// file, which if successful will then be used for the next call to Apply.
+func (wd *WorkingDir) CreatePlan(ctx context.Context) (CreatePlanResponse, error) {
+	tfJSON := NewTerraformJSONBuffer()
+	createPlanResponse := CreatePlanResponse{}
+
+	err := wd.CreatePlanJSON(ctx, tfJSON)
+	target := &tfexec.ErrVersionMismatch{}
+	if !errors.As(err, &target) {
+		createPlanResponse.Diagnostics = tfJSON.Diagnostics()
+		createPlanResponse.Stdout = tfJSON.RawOutput()
+
+		return createPlanResponse, err
+	}
+
 	logging.HelperResourceTrace(ctx, "Calling Terraform CLI plan command")
 
 	hasChanges, err := wd.tf.Plan(context.Background(), tfexec.Reattach(wd.reattachInfo), tfexec.Refresh(false), tfexec.Out(PlanFileName))
 
 	logging.HelperResourceTrace(ctx, "Called Terraform CLI plan command")
 
+	if err != nil {
+		return createPlanResponse, err
+	}
+
+	if !hasChanges {
+		logging.HelperResourceTrace(ctx, "Created plan with no changes")
+
+		return createPlanResponse, nil
+	}
+
+	stdout, err := wd.SavedPlanRawStdout(ctx)
```

Comment:
Should this `stdout` be wired with the `CreatePlanResponse` stdout? Maybe the overlappting naming is confusing me. 😄

Reply:
I have removed the usage of stdout from `CreatePlanResponse` and the other `...Response`(s) as it does not seem to be necessary or desirable to show this level of information to provider developers.